packages/alfa-style/src/style.ts

```diff
@@ -52,6 +54,18 @@ export class Style implements Serializable {
     this._device = device;
     this._parent = parent;
 
+    // First pass: Resolve cascading variables which will be used in the second
+    // pass.
+    for (const declaration of declarations) {
+      const { name, value } = declaration;
+
+      if (name.startsWith("--") && !this._variables.has(name)) {
+        this._variables.set(name, Lexer.lex(value));
+      }
```

Comment:
> Might be OK because I `declarations` is maybe already in reverse order, I don't remember fully…

Exactly 👍 The declarations are listed from most specific to least specific so `--foo: red` would be encountered first.